lib/internal/Magento/Framework/GraphQl/Query/EnumLookup.php

```diff
@@ -43,23 +44,19 @@ public function __construct(ConfigInterface $typeConfig, DataMapperInterface $en
      * @param string $enumName
      * @param string $fieldValue
      * @return string
-     * @throws \Magento\Framework\Exception\RuntimeException
      */
     public function getEnumValueFromField(string $enumName, string $fieldValue) : string
     {
-        $priceViewEnum = $this->typeConfig->getConfigElement($enumName);
-        if ($priceViewEnum instanceof Enum) {
-            foreach ($priceViewEnum->getValues() as $enumItem) {
-                $mappedValues = $this->enumDataMapper->getMappedEnums($enumName);
-                if (isset($mappedValues[$enumItem->getName()]) && $mappedValues[$enumItem->getName()] == $fieldValue) {
-                    return $enumItem->getValue();
-                }
+        /** @var Enum $enumObject */
+        $enumObject = $this->typeConfig->getConfigElement($enumName);
+        $mappedValues = $this->enumDataMapper->getMappedEnums($enumName);
+
+        foreach ($enumObject->getValues() as $enumItem) {
+            if (isset($mappedValues[$enumItem->getName()]) && $mappedValues[$enumItem->getName()] == $fieldValue) {
+                return $enumItem->getValue();
             }
-        } else {
-            throw new \Magento\Framework\Exception\RuntimeException(
```

Comment:
What is the reason to remove exception here?

Reply:
Because it will never happen.

See, [\Magento\Framework\GraphQl\ConfigInterface](https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/GraphQl/ConfigInterface.php#L25)

Reply:
@atwixfirster As far as I understood `Magento\Framework\GraphQl\ConfigInterface::getConfigElement` returns `ConfigElementInterface` and that is not necessary `Enum`, so that may happen. `Enum` does extends `ConfigElementInterface` but for your point to be valid the `Magento\Framework\GraphQl\ConfigInterface::getConfigElement` should explicit return the `Enum` interface.

Please let me know your thoughts.

Reply:
agreed, @gabrieldagama 

**fixed**